Rokt.Widget/android/src/main/java/com/rokt/reactnativesdk/RNRoktWidgetModule.kt

```
@@ -44,23 +46,15 @@ class RNRoktWidgetModule internal constructor(private val reactContext: ReactApp
 
     @ReactMethod
     fun initialize(roktTagId: String?, appVersion: String?) {
-        val currentActivity = currentActivity
-        if (currentActivity != null && appVersion != null && roktTagId != null) {
-            Rokt.setFrameworkType(ReactNative)
-            Rokt.init(roktTagId, appVersion, currentActivity)
-        } else {
-            logDebug("Activity, roktTagId and AppVersion cannot be null")
+        initRokt(roktTagId, appVersion, 4) { activity ->
+            Rokt.init(requireNotNull(roktTagId), requireNotNull(appVersion), activity)
```

Any concerns with changing behaviour to throw NPE instead of logging if any are null?

---

This shouldn't throw an NPE. `initRokt` actually verifies the `roktTagId` and `appVersion` and won't invoke the lambda. We could have passed the non-null `roktTagId` and `appVersion` from the lambda to make it clean, but didn't want to make it more complex.

---

Ah yes completely I missed the first lines of initRokt 😝 makes sense